docs/security/self-assessment.md

```diff
@@ -0,0 +1,167 @@
+3. **SyncController:** Periodically synchronize cloud-side resources to ensure consistency of cloud-side data.
+4. **DynamicController:** Based on K8S Dynamic Client to monitor K8S resources, realize resource management and edge node list&watch capability.
+5. **DeviceController:** Device and Device Model resources are used to manage devices and achieve terminal device management.
+
+**In Edge-Core:**
+
+1. **EdgeHub**: A websocket client for CloudHub's websocket server responsible for sending/receiving messages of/for all edge components to/from Cloud.
+2. **Edged:** An agent that runs on edge nodes and manages containerized applications. It is a shrinked kubelet for IOT/Edge following the same K8S types and APIs. 
+3. **MetaManager:** the message processor between edged and edgehub. It is also responsible for storing/retrieving metadata to/from a lightweight database (SQLite). It contains MetaServer module, which realizes edge node to provide list&Watch capability externally.
+4. **DeviceTwin:** The module responsible for storing device status and syncing device status to the cloud. It also provides query interfaces for applications. It acts as a mirror to actual device metadata.
+5. **EventBus:** A MQTT client to interact with MQTT servers (mosquitto), offering publish and subscribe capabilities to other components.
+6. **ServiceBus:** ServiceBus is an HTTP client running on the edge. It receives requests from cloud services and interacts with the HTTP server running on the edge, enabling cloud services to access the HTTP server running on the edge over HTTP.
+
+## Self-assessment use
+
+This self-assessment is created by the KubeEdge team to perform an internal analysis of the project's security.  It is not intended to provide a security audit of KubeEdge, or function as an independent assessment or attestation of KubeEdge's security health.
+
+This document serves to provide KubeEdge users with an initial understanding of KubeEdge's security, where to find existing security documentation, KubeEdge plans for security, and general overview of KubeEdge security practices, both for development of KubeEdge as well as security of KubeEdge.
+
+This document provides the CNCF TAG-Security with an initial understanding of KubeEdge to assist in a joint-review, necessary for projects under incubation.  Taken together, this document and the joint-review serve as a cornerstone for if and when KubeEdge seeks graduation and is preparing for a security audit.
+
+## Security functions and features
+
+* Critical.  The cloud component CloudHub exposes ports to provide external services, accept connections from edge nodes, and be responsible for cloud-side communication. During communication, webSocket or Quick protocol is used for data transmission, and TLS certificate is used for data encryption.
+* Critical.  The EdgeHub component exposes the port to provide services externally, and provides list&Watch capability externally on the edge side.
```

Comment:
This paragraph is confusing, I think we should list a set of security relevant functions and features. E.g. what we did to protect from security threat.